client-spark/common/src/main/scala/org/apache/celeborn/spark/FailedShuffleCleaner.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.celeborn.spark
+import java.util
+import java.util.concurrent.{ConcurrentHashMap, LinkedBlockingQueue}
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+import org.apache.spark.scheduler.{RunningStageManager, RunningStageManagerImpl}
+
+import org.apache.celeborn.client.LifecycleManager
+import org.apache.celeborn.common.internal.Logging
+
+private[celeborn] object FailedShuffleCleaner extends Logging {
+
+  private val lifecycleManager = new AtomicReference[LifecycleManager](null)
+  // in celeborn ids
+  private val shufflesToBeCleand = new LinkedBlockingQueue[Int]()
+  private val cleanedShuffleIds = new mutable.HashSet[Int]
+  // celeborn shuffle id to stage id referred to it
+  private[celeborn] val celebornShuffleIdToReferringStages =
+    new ConcurrentHashMap[Int, mutable.HashSet[Int]]()
```

Comment:
```suggestion
    ConcurrentHashMap.newKeySet[Int]()
```
This can remove the synchronized block.

Reply:
We have a misunderstanding here, I mean that you should use ConcurrentHashMap.newKeySet[Int] as the value type of the ConcurrentHashMap. This could remove the synchronization block.

`new ConcurrentHashMap[Int,ConcurrentHashMap[Int,Boolean]]`

Reply:
actually I found I still need an explicit lock there , e.g. https://github.com/apache/celeborn/pull/3109/files#diff-b7440adf323f1a6c564d55c9c18a414c0a1a2039b5babf1657172c6d0c68c17fR144-R157 operations here cannot be thread safe only with ConcurrentHashMap.KeyView,